payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardDetailsSectionElementUI.kt

```diff
@@ -0,0 +1,111 @@
+package com.stripe.android.ui.core.elements
+
+import android.content.Intent
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RestrictTo
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.stripe.android.stripecardscan.cardscan.CardScanSheetResult
+import com.stripe.android.stripecardscan.cardscan.exception.UnknownScanException
+import com.stripe.android.ui.core.DefaultIsStripeCardScanAvailable
+import com.stripe.android.ui.core.PaymentsTheme
+import com.stripe.android.ui.core.R
+import com.stripe.android.ui.core.cardscan.CardScanActivity
+
+@Composable
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun CardDetailsSectionElementUI(
+    enabled: Boolean,
+    controller: CardDetailsSectionController,
+    hiddenIdentifiers: List<IdentifierSpec>?
+) {
+    val context = LocalContext.current
+    val isStripeCardScanAvailable = DefaultIsStripeCardScanAvailable()
+
+    Row(
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = 9.dp),
+    ) {
+        H6Text(
+            text = stringResource(R.string.card_information),
+            modifier = Modifier
+                .semantics(mergeDescendants = true) { // Need to prevent form as focusable accessibility
+                    heading()
+                }
+        )
+        if (isStripeCardScanAvailable()) {
+            val cardScanLauncher =
+                rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+                    it.data?.let {
+                        controller.cardDetailsElement.controller.numberElement.controller.onCardScanResult(
```

Comment:
This line feels a little weird. Maybe this logic should be moved to your controller class?

Reply:
I'm not sure how I would move this logic - the rememberLauncherForActivityResult method is a Composable method, and I'm calling the onCardScanResult function in the controller from the callback

Reply:
oh fair enough. You can put a @Composable method in the controller if you want. The bit that I'm feeling weird about is the `controller.cardDetailsElement.controller.numberElement.controller.onCardScanResult` but I think that's required weirdness right? We couldn't hide that away in the controller? I like not having a ton of logic in the compose views and making them as dumb as possible.

Reply:
An alternative would be to add a function in the CardDetailsSectionElementController that then calls the `cardDetailsElement.controller.numberElement.controller.onCardScanResult()` function